src/Classes/RichTextViewController.swift

```diff
@@ -435,17 +484,233 @@ public class RichTextViewController: UIViewController {
+                range.location -= 1
+            }
+        }
+        var loops = 0
+        var testString = (textView.text as NSString).substringWithRange(range)
+        while loops < 2 {
+            if testString == bulletedLineStarter {
+                range.location += 2
+                range.length = 0
+                textView.selectedRange = range
+                break
+            }
+            
+            if range.location > 0 {
+                range.location -= 1
+            } else {
+                break
+            }
+            testString = (textView.text as NSString).substringWithRange(range)
+            loops += 1
+        }
+        
+    }
+    
+    public func toggleBulletedList() {
+        if selectionContainsNumberedList(textView.selectedRange) {
+            toggleNumberedList()
+        }
+        
+        if textView.selectedRange.length == 0 {
+            if selectionContainsBulletedList(textView.selectedRange) {
+                if let bulletIndex = textView.text.previousIndexOfSubstring(bulletedLineStarter, fromIndex: textView.selectedRange.location) {
+                    removeTextFromRange(NSRange(location: bulletIndex, length: bulletedLineStarter.length), fromTextView: textView)
+                }
+            } else {
+                let newLineIndex = (textView.text.previousIndexOfSubstring("\n", fromIndex: textView.selectedRange.location) ?? -1) + 1
+                addText(bulletedLineStarter, toTextView: textView, atIndex: newLineIndex)
+            }
+        } else {
+            var bulletsInSelection = false
+            
+            if selectionContainsBulletedList(NSRange(location: textView.selectedRange.location, length: 0)), let bulletIndex = textView.text.previousIndexOfSubstring(bulletedLineStarter, fromIndex: textView.selectedRange.location) {
+                bulletsInSelection = true
+                removeTextFromRange(NSRange(location: bulletIndex, length: bulletedLineStarter.length), fromTextView: textView)
+            }
+            
+            if selectionContainsBulletedList(textView.selectedRange) {
+                bulletsInSelection = true
+                var index = textView.selectedRange.location
+                while true {
+                    guard let nextBulletIndex = textView.text.nextIndexOfSubstring(bulletedLineStarter, fromIndex: index)  else { break }
+
+                    removeTextFromRange(NSRange(location: nextBulletIndex, length: bulletedLineStarter.length), fromTextView: textView)
+                    index = nextBulletIndex
+                }
+            }
+            
+            if !bulletsInSelection {
+                let newLineIndex = (textView.text.previousIndexOfSubstring("\n", fromIndex: textView.selectedRange.location) ?? -1) + 1
+                addText(bulletedLineStarter, toTextView: textView, atIndex: newLineIndex)
+                
+                var index = textView.selectedRange.location
+                while true {
```

Comment:
This look scares me. It screams, infinite loop. Is there a safer way to do this?